adoc/chapters/programming_interface.adoc

```diff
@@ -6959,6 +6959,35 @@ If this is a <<ranged-accessor>>, returns the offset that was specified when the
 accessor was constructed.  For other accessors, returns the default constructed
 [code]#id<Dimensions>{}#.
 
+a@
+[source]
+----
+global_ptr<access::decorated::legacy> get_pointer() const noexcept
+----
+   a@ Available only when [code]#(AccessTarget == target::device)#.
+
+Returns a [code]#multi_ptr# to the start of this accessor's underlying buffer,
+even if this is a <<ranged-accessor>> whose range does not start at the
+beginning of the buffer.  The return value is unspecified if the accessor is
+empty.
+
+This function may only be called from within a <<command>>.
+
+Deprecated in SYCL 2020.  Use [code]#get_multi_ptr# instead.
+
+a@
+[source]
+----
+std::add_pointer_t<value_type> get_pointer() const noexcept
+----
+   a@ Available only when [code]#(AccessTarget == target::host_task)#.
+
+Returns a pointer to the start of this accessor's underlying buffer, even if
+this is a <<ranged-accessor>> whose range does not start at the beginning of
+the buffer.  The return value is unspecified if the accessor is empty.
+
+This function may only be called from within a <<command>>.
+
 a@
 [source]
 ----
```

Comment:
While reviewing this, I noticed that there is a bug in the table entry below this comment.  The `get_multi_ptr` member function should be available only when `(AccessTarget == target::device)`.  We do not want to make a `multi_ptr` available to code running in a host task.

We could either include that fix as part of this PR, or create a new PR to fix it.

If you do fix it in this PR, remember also to add an "available only when" comment to the synopsis.

Reply:
Yeah that's a good point, I was thinking about this when I was making this change, we say that `accessor_ptr` is unspecified when targeting `host_trask` but we don't restrict `get_multi_ptr` from being defined, I think it makes sense to do this.

I've created a follow-up PR to address this: https://github.com/KhronosGroup/SYCL-Docs/pull/437.